includes/gateway/class-omise-payment-paynow.php

```diff
@@ -70,16 +83,19 @@ public function display_qrcode( $order, $context = 'view' ) {
 		$charge_id = $this->get_charge_id_from_order();
 		$charge    = OmiseCharge::retrieve( $charge_id );
 		$qrcode    = $charge['source']['scannable_code']['image']['download_uri'];
-		?>
-		<div class="omise omise-paynow-details" <?php echo 'email' === $context ? 'style="margin-bottom: 4em; text-align:center;"' : ''; ?>>
-			<div class="omise omise-paynow-logo"></div>
-			<p>
-				<?php echo __( 'Scan the QR code to pay', 'omise' ); ?>
-			</p>
-			<div class="omise omise-paynow-qrcode">
-				<img src="<?php echo $qrcode; ?>" alt="Omise QR code ID: <?php echo $charge['source']['scannable_code']['image']['id']; ?>">
+
+		if ( 'view' === $context ) : ?>
```

Comment:
We should exclude this part if order placement has failed.

Reply:
Am I understanding correctly that you mean when buyer attempts to scan the QR code and get `failed` status back from Charge API?

Reply:
sorry for the confusion, I wanted to add this comment into other part. consider this for line no 97.